### PR TITLE
Bug fixing: Json decoder extracts wrong time field value if schema contains outgoingTimeFieldSpec

### PR DESCRIPTION
**Pull request by fx19880617**
Current Json decoder doesn't properly handle time field value extraction when configure outgoing time spec.

Should extract value based on incoming field name.

Thanks to Alex Filipchik for reporting this bug.

### CODE REVIEW

File: pinot-connectors/pinot-connector-kafka-base/src/main/java/org/apache/pinot/core/realtime/impl/kafka/KafkaJSONMessageDecoder.java [hunk 1]
```diff
@@ -33,21 +35,30 @@
...
     try {
       JsonNode message = JsonUtils.bytesToJsonNode(payload);
-      for (FieldSpec fieldSpec : schema.getAllFieldSpecs()) {
-        String column = fieldSpec.getName();
-        destination.putField(column, JsonUtils.extractValue(message.get(column), fieldSpec));
+      for (FieldSpec fieldSpec : _schema.getAllFieldSpecs()) {
+        FieldSpec incomingFieldSpec =
+            fieldSpec.getFieldType() == FieldSpec.FieldType.TIME ? _incomingTimeFieldSpec : fieldSpec;
+        String column = incomingFieldSpec.getName();
+        destination.putValue(column, JsonUtils.extractValue(message.get(column), fieldSpec));
```
Comment (Jackie-Jiang) [inline pinot-connectors/pinot-connector-kafka-base/src/main/java/org/apache/pinot/core/realtime/impl/kafka/KafkaJSONMessageDecoder.java]: `incomingFieldSpec`?

File: pinot-connectors/pinot-connector-kafka-base/pom.xml [hunk 2]
```diff
@@ -35,4 +35,19 @@
...
+
+  <dependencies>
+    <!-- test -->
+    <dependency>
+      <groupId>org.testng</groupId>
+      <artifactId>testng</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.google.code.gson</groupId>
```
[the remark this answers is not in the document]
Comment (fx19880617) [reply pinot-connectors/pinot-connector-kafka-base/pom.xml]: will do

File: pinot-connectors/pinot-connector-kafka-base/src/test/java/org/apache/pinot/core/realtime/impl/kafka/KafkaJSONMessageDecoderTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      throws Exception {
+    KafkaJSONMessageDecoder decoder = new KafkaJSONMessageDecoder();
+    decoder.init(new HashMap<>(), schema, "testTopic");
+    GenericRow r = new GenericRow();
+    String line = reader.readLine();
+    while (line != null) {
+      JsonObject jsonObject = JsonParser.parseString(line).getAsJsonObject();
+      decoder.decode(line.getBytes(), r);
+      for (FieldSpec fieldSpec : schema.getAllFieldSpecs()) {
+        String fieldSpecName = (fieldSpec.getFieldType() == FieldSpec.FieldType.TIME) ? schema.getTimeFieldSpec()
```
[the remark this answers is not in the document]
Comment (fx19880617) [reply pinot-connectors/pinot-connector-kafka-base/src/test/java/org/apache/pinot/core/realtime/impl/kafka/KafkaJSONMessageDecoderTest.java]: good catch

File: pinot-connectors/pinot-connector-kafka-base/src/test/java/org/apache/pinot/core/realtime/impl/kafka/KafkaJSONMessageDecoderTest.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+  private static ObjectMapper objectMapper = new ObjectMapper();
+
+  @Test
+  public void testJsonDecoderWithoutOutgoingTimeSpec()
+      throws Exception {
+    Schema schema = Schema.fromFile(new File(
+        getClass().getClassLoader().getResource("data/test_sample_data_schema_without_outgoing_time_spec.json")
+            .getFile()));
+    BufferedReader reader = new BufferedReader(
```
Comment (Jackie-Jiang) [inline pinot-connectors/pinot-connector-kafka-base/src/test/java/org/apache/pinot/core/realtime/impl/kafka/KafkaJSONMessageDecoderTest.java]: Can we make this try-with-resource and remove the `reader.close()` in `testJsonDecoder()`? Same for other tests.

Comment (fx19880617) [reply pinot-connectors/pinot-connector-kafka-base/src/test/java/org/apache/pinot/core/realtime/impl/kafka/KafkaJSONMessageDecoderTest.java]: do you mean something like: 
```
    try (BufferedReader reader = new BufferedReader(
        new FileReader(getClass().getClassLoader().getResource("data/test_sample_data.json").getFile()))) {
      testJsonDecoder(schema, reader);
    }

**Comment by zen201415** [comment]
This seems like a serious problem, why don't you merge it into 0.2.0?

**Pull request merged**